shared/src/simulation/behaviors/assign-leader.ts

```diff
@@ -35,51 +42,189 @@ export const assignLeaderBehavior: SimulationBehavior<AssignLeaderBehaviorState>
+                                        isInSpecificSimulatedRegion(
+                                            person,
+                                            simulatedRegion.id
+                                        )
+                                    );
+                                    const groupedPersonnel = groupBy(
+                                        personnel,
+                                        (person) => person.personnelType
+                                    );
+                                    personnelCount.gf =
+                                        groupedPersonnel['gf']?.length ?? 0;
+                                    personnelCount.notSan =
+                                        groupedPersonnel['notSan']?.length ?? 0;
+                                    personnelCount.notarzt =
+                                        groupedPersonnel['notarzt']?.length ??
+                                        0;
+                                    personnelCount.rettSan =
+                                        groupedPersonnel['rettSan']?.length ??
+                                        0;
+                                    personnelCount.san =
+                                        groupedPersonnel['san']?.length ?? 0;
+                                }
+                                break;
+                            case 'vehicleCount':
+                                {
+                                    const radiogram = getActivityById(
+                                        draftState,
+                                        simulatedRegion.id,
+                                        event.generateReportActivityId,
+                                        'generateReportActivity'
+                                    )
+                                        .radiogram as Mutable<VehicleCountRadiogram>;
+                                    const vehicles = Object.values(
+                                        draftState.vehicles
+                                    ).filter((vehicle) =>
+                                        isInSpecificSimulatedRegion(
+                                            vehicle,
+                                            simulatedRegion.id
+                                        )
+                                    );
+                                    const groupedVehicles = groupBy(
+                                        vehicles,
+                                        (vehicle) => vehicle.vehicleType
+                                    );
+                                    Object.entries(groupedVehicles).forEach(
+                                        ([vehicleType, vehicleGroup]) => {
+                                            radiogram.vehicleCount[
+                                                vehicleType
+                                            ] = vehicleGroup.length;
+                                        }
```

Comment:
You could also use `map` instead of `forEach`. I think this would make it a bit more clear what is going on.

Reply:
Since this executes side effects I refrained from doing so

Reply:
I don't really see what side effects would be executed by something like this. Do you mind explaining (maybe as dm)
```ts
radiogram.vehicleCount = Object.fromEntries(
    StrictObject.entries(
        groupedVehicles
    ).map(([vehicleType, vehicleGroup]) => [
        vehicleType,
        vehicleGroup.length,
    ])
);
```

Reply:
Done